ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -5,17 +5,29 @@
 #include <OS/log.h>
 #include <IOKit/IOKitLib.h>
 #include <signal.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+using std::string;
+using std::to_string;
 
 static const char PrjFSKextLogDaemon_OSLogSubsystem[] = "org.vfsforgit.prjfs.PrjFSKextLogDaemon";
+static const int INVALID_SOCKET_FD = -1;
 
 static os_log_t s_daemonLogger, s_kextLogger;
 static IONotificationPortRef s_notificationPort;
 
+static int s_messageListenerSocket = INVALID_SOCKET_FD;
+static string s_messageListenerSocketPath = "/usr/local/GitService/pipe/vfs-c780ac06-135a-4e9e-ab6c-d41e2d265baa";
```

Comment:
Should this be in a config, and should that change be part of this PR?

It would be a bit of a hack but we also have the option of reading this from `GIT_TR2_EVENT`.

/cc: @jrbriggs @jeffhostetler for their thoughts

Reply:
Just leave it for now or #define it somewhere.  Or see whe Matthew is doing on the mount side.

Don’t use the GIT_TR2_EVENT. It points to a different pipe.  And I’m deprecating it in favor of a system config setting that’ll be easier for the Mac installers to set.

Reply:
> see whe Matthew is doing on the mount side.

It looks like on the mount side we're shelling out to git and looking for the `gvfs.telemetry-pipe` config value (in the global or system git config).

> in favor of a system config setting that’ll be easier for the Mac installers to set.

This makes me wonder if all three applications (Git, VFS4G, PrjFSKextLogDaemon) should use the same config setting or each have their own.  In the case of PrjFSKextLogDaemon there no reason the config setting would have to be in a git config, it could be in a local PrjFSKextLogDaemon specific config file as well.

I guess it partially comes down to if we expect PrjFSKextLogDaemon to be used on machines without Git/VFS4G.  In the short to medium term I think the answer is that we don't expect \ scenarios like that (unlike Windows, ProjFS on Mac is not currently a standalone product).

/cc: @jrbriggs @mjcheetham

Reply:
I’m constrained on git.exe because of core Git on where I can get the value, but the kext and mount processes could share or not share config data as you want.  And yeah, requiring the mount daemon spawn a child process to get a config value seems expensive.

The git-guid and vfs-guid pipes are different.  The former summarizes all events for the client whereas the latter is just a proxy server. And the have a different protocol, so we can’t blend them. So I need to keep git.exe setting separate anyway.

Reply:
Per offline discussion with @jrbriggs we'll leave `s_messageListenerSocketPath` hardcoded for now.